x-pack/plugins/discover_log_explorer/public/utils/dataset_selection/hydrate_dataset_selection.ts.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AllDatasetSelection } from './all_dataset_selection';
+import { SingleDatasetSelection } from './single_dataset_selection';
+import { DatasetSelectionPlain } from './types';
+
+export const hydrateDatasetSelection = (datasetSelection: DatasetSelectionPlain) => {
```

Comment:
This is added in preparation for the following task, where we'll need to transform it into a DatasetSelection instance once the URL state is correctly restored.